### ISSUE TITLE
`BIGINT PRIMARY KEY AUTOINCREMENT` in SQLite

### ISSUE THREAD

**Issue by tyt2y3** [SeaQL/sea-query#689]
https://sqlite.org/forum/info/2dfa968a702e1506e885cb06d92157d492108b22bf39459506ab9f7125bca7fd

SQLite would give `error: "AUTOINCREMENT is only allowed on an INTEGER PRIMARY KEY"`

So I guess we somehow have to make `BIGINT PRIMARY KEY AUTOINCREMENT` a special case and convert it implicitly to `INTEGER PRIMARY KEY AUTOINCREMENT`

Ref: https://github.com/SeaQL/sea-orm/issues/1832

**Comment by Perksey** [comment]
Yes, just hit this.

```
thread 'backend::tests::sqlite_playground' panicked at 'called `Result::unwrap()` on an `Err` value: SQLite(SqliteFailure(Error { code: Unknown, extended_code: 1 }, Some("AUTOINCREMENT is only allowed on an INTEGER PRIMARY KEY")))', src/backend/mod.rs:288:55
```

For generated query

```sql
CREATE TABLE IF NOT EXISTS "example_table" ( "id" bigint PRIMARY KEY AUTOINCREMENT, "my_field1" integer NOT NULL, "my_field2" bigint )
```

I would personally suggest that sea-query maps the backend's type affinity exactly i.e. always mapping the sized integer types to just `INTEGER`, mapping both float and double to `REAL`, etc. Really sea-query is an abstraction over multiple SQL languages, and while it's very useful to the user to hint sized types for backends that do support more rigid type affinity, its usefulness wears off when the abstraction tries to preserve this rigidity for a database/query language that fundamentally isn't modelled that way and does not support/do anything with the information it is being given.

I'm happy to do the work in a PR if this is agreeable.

**Comment by tyt2y3** [comment]
A PR on the error above would be appreciated.
However I am not sure I understand your preference to type mapping. Or I think we are having the opposite. But only because we wanted to preserve the *intended type*, such that we when doing schema discovery, we'd get back `i32` vs `i64`. But that's more an ORM thing.

**Comment by tyt2y3** [comment]
```rust
pub struct SqliteQueryBuilder<TTypeMapper: SqliteTypeMapper = ExactMapper>;
```

This is a great idea

**Comment by tyt2y3** [comment]
Fixed via 37a30f6cc8dcd9a61c80bbc6ff8ff249831e2129
